packages/core/src/lib/icon/icons.ts

```diff
@@ -76,3 +79,12 @@ export const paths = {
  * The possible icon names that can be rendered. This is good for typing props.
  */
 export type IconName = keyof typeof paths;
+
+const ICON_NAMES = Object.keys(paths) as IconName[];
+
+/**
+ * A map of icon names, useful for dynamically adding icons with type safety.
+ */
+export const iconNames = Object.fromEntries<IconName>(
+  ICON_NAMES.map((name) => [name, name])
+) as Record<IconName, IconName>;
```

Comment:
Open to suggestions on a better way of easily using icon names dynamically, this came up in the radio and I know we do icon selection elsewhere. Usually that looks something like:

let icon: IconName:
$: {
  if (thing) {
    icon = 'some-magic-string';
  } elseif {
    icon = 'another-magic-string';
  } else {
    icon = 'yet-another-magic-string';
  }
}
Don't love having all the magic strings, and while that works for singular icons it gets messier with determining icons for multiple elements like in the radio. This at least gives us a quick map of names we can refer to, and should make refactoring icon names slightly less painful in the future. If we like this (or something else that accomplishes the same thing), I can refactor other places where we declare icon names to use it as well for consistency.

Reply:
> If we like this (or something else that accomplishes the same thing)

I'm unclear on what exactly this export is accomplishing, based on usage in this PR. Would you mind explaining a little more?

```ts
$: getIcon = (option: string): IconName =>
  isSelected(option)
    ? iconNames['radiobox-marked']
    : iconNames['radiobox-blank'];
```

doesn't seem very different, magic-strings-wise, than:

```ts
$: getIcon = (option: string): IconName =>
  isSelected(option)
    ? 'radiobox-marked'
    : 'radiobox-blank';
```

Reply:
So I actually think this was a case of vscode getting confused. I originally tried

```
$: getIcon = (option: string): IconName =>
  isSelected(option)
    ? 'radiobox-marked'
    : 'radiobox-blank';
```

And I was getting an error that I could not cast a `string` to an `IconName`. I thought this was an issue where it was not properly interpreting the type from the strings, but having restarted vscode and going to snap a screenshot of the error, it is no longer showing. I think the reason was, I added those two icons to `paths`, but in the `radio` component it was not detecting that I had added them, so it was just interpreting them as a `string`.

I feel like I need to put a sticky-note on my screen that says "Restart eslint/svelte server in vscode before debugging type issues." It comes up more than I would like.